tests/kit/model_zoo/registry.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+from dataclasses import dataclass
+from typing import Callable
+
+__all__ = ['ModelZooRegistry', 'ModelAttributem', 'model_zoo']
+
+
+@dataclass
+class ModelAttribute:
+    """
+    Attributes of a model.
+    """
+    has_control_flow: bool = False
+
+
+class ModelZooRegistry(dict):
+    """
+    A registry to map model names to model and data generation functions.
+    """
+
+    def register(self,
+                 name: str,
+                 model_fn: Callable,
+                 data_gen_fn: Callable,
+                 output_transform_fn: Callable,
+                 model_attribute: ModelAttribute = None):
```

Comment:
How to set model init args? By this `model_attribute` or overwriting `model_fn`?

Reply:
@FrankLeeeee I agree with @ver217. We should add config_kwargs for every type of model.

Reply:
> 

`model_attribute` is used for logic checks, for example, you may want to differentiate models with/without control flow in tracing/auto parallel. If you want to set init args when registering. You can do:

```
model_fn = lambda: model_cls(arg1=x, arg2=y)
```